src/compositor/wayland_wrapper/qwlkeyboard.cpp

```diff
@@ -327,23 +324,38 @@ void Keyboard::createXKBKeymap()
                                          strdup(qPrintable(m_keymap.options())) };
     struct xkb_keymap *keymap = xkb_keymap_new_from_names(m_context, &rule_names, static_cast<xkb_keymap_compile_flags>(0));
 
+    if (!keymap) {
+        qWarning("Failed to load the '%s' XKB keymap.", qPrintable(m_keymap.layout()));
+        return;
+    }
+
     char *keymap_str = xkb_keymap_get_as_string(keymap, XKB_KEYMAP_FORMAT_TEXT_V1);
-    if (!keymap_str)
-        qFatal("Failed to compile global XKB keymap");
+    if (!keymap_str) {
+        qWarning("Failed to compile global XKB keymap");
+        xkb_keymap_unref(keymap);
+        return;
+    }
 
     m_keymap_size = strlen(keymap_str) + 1;
     m_keymap_fd = createAnonymousFile(m_keymap_size);
-    if (m_keymap_fd < 0)
-        qFatal("Failed to create anonymous file of size %lu", static_cast<unsigned long>(m_keymap_size));
+    if (m_keymap_fd < 0) {
+        qWarning("Failed to create anonymous file of size %lu", static_cast<unsigned long>(m_keymap_size));
+        xkb_keymap_unref(keymap);
```

Comment:
Wonder if this and following should keep qFatal(), don't look like those should trigger at least from keymap loading failure.

Btw, existing issue, but looks like previous m_keymap_fd is not close()d if setting a new layout. Should?